### ISSUE TITLE
Doesn't work with nodejs

### ISSUE THREAD

**Issue by xlc** [paritytech/substrate-connect#1145]
Code: https://github.com/xlc/substrate-connect-jest

Running directly with ts-node will error with 

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/xiliangchen/projects/acala/sc-test/node_modules/@substrate/connect/dist/index.js from /Users/xiliangchen/projects/acala/sc-test/node_modules/@polkadot/rpc-provider/cjs/substrate-connect/ScProvider.js not supported.
Instead change the require of index.js in /Users/xiliangchen/projects/acala/sc-test/node_modules/@polkadot/rpc-provider/cjs/substrate-connect/ScProvider.js to a dynamic import() which is available in all CommonJS modules.
    at Object.require.extensions.<computed> [as .js] (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/index.js:842:20)
    at Object.<anonymous> (/Users/xiliangchen/projects/acala/sc-test/node_modules/@polkadot/rpc-provider/cjs/substrate-connect/ScProvider.js:16:16)
    at Object.require.extensions.<computed> [as .js] (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/index.js:842:20)
    at Object.<anonymous> (/Users/xiliangchen/projects/acala/sc-test/node_modules/@polkadot/rpc-provider/cjs/substrate-connect/index.js:7:19)
    at Object.require.extensions.<computed> [as .js] (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/index.js:842:20)
    at Object.<anonymous> (/Users/xiliangchen/projects/acala/sc-test/src/index.ts:3:29)
    at Module.m._compile (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/index.js:848:29)
    at Object.require.extensions.<computed> [as .ts] (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/index.js:850:16)
    at phase4 (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/bin.js:414:16)
    at bootstrap (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/bin.js:49:12)
    at main (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/bin.js:32:12)
    at Object.<anonymous> (/Users/xiliangchen/projects/acala/sc-test/node_modules/ts-node/dist/bin.js:526:5) {
  code: 'ERR_REQUIRE_ESM'
```

Running with jest will SIGSEGV and maybe related to https://github.com/nodejs/node/issues/35889

Can we have a version that doesn't use `import`? Why it is even needed at the first place? To reduce initial JS bundle size? But the app should be able to import the substrate connect package instead. 

I only want to run it as a nodejs server and it just doesn't work.

**Comment by wirednkod** [comment]
Thank you @xlc .
This is already [on the way](https://github.com/paritytech/substrate-connect/pull/1142). [This](https://github.com/paritytech/smoldot/pull/2487) and [this](https://github.com/paritytech/substrate-connect/pull/1143) need first to merge and then the #1142 will be able to proceed

**Comment by tomaka** [comment]
> Why it is even needed at the first place?

`import` in JavaScript is the way forward. It's been decided years and years ago by the JS ecosystem.
NodeJS v12 and further support this feature by default since April/May 2020. Versions below support it if you pass the `--experimental-modules` flag.

**Comment by xlc** [comment]
But it still doesn't work with jest 💁‍♂️

**Comment by tomaka** [comment]
Should be fixed in substrate-connect v0.7.9
